diff --git a/cmake/LibraryDefine.cmake b/cmake/LibraryDefine.cmake
index f4ec324..5fbc95c 100644
--- a/cmake/LibraryDefine.cmake
+++ b/cmake/LibraryDefine.cmake
@@ -58,7 +58,7 @@ function(OPENEXR_DEFINE_LIBRARY libname)
   else()
       target_compile_definitions(${objlib} PUBLIC OPENEXR_USE_DEFAULT_VISIBILITY)
   endif()
   if (_openexr_extra_flags)
-    target_compile_options(${objlib} PUBLIC ${_openexr_extra_flags})
+    target_compile_options(${objlib} PRIVATE ${_openexr_extra_flags})
   endif()
   set_property(TARGET ${objlib} PROPERTY PUBLIC_HEADER ${OPENEXR_CURLIB_HEADERS})
